crates/polars-plan/src/plans/conversion/type_coercion/mod.rs

```diff
@@ -337,6 +337,48 @@ impl OptimizationRule for TypeCoercionRule {
                 let function = function.clone();
                 let mut input = input.clone();
 
+                // Do some type enforcement and handling of edge cases for
+                // search_sorted() in particular.
+                #[cfg(feature = "search_sorted")]
+                if matches!(function, FunctionExpr::SearchSorted(_)) {
+                    let (_, type_left) = unpack!(get_aexpr_and_type(
+                        expr_arena,
+                        input[0].node(),
+                        &input_schema
+                    ));
+                    let (_, type_other) = unpack!(get_aexpr_and_type(
+                        expr_arena,
+                        input[1].node(),
+                        &input_schema
+                    ));
+
+                    let left_nl = type_left.nesting_level();
+                    let right_nl = type_other.nesting_level();
+                    polars_ensure!(
+                        type_other.is_null() || (right_nl >= left_nl && (right_nl - left_nl) <= 1),
+                        InvalidOperation: "wrong nesting depth {type_left} searching for {type_other}"
+                    );
+
+                    // Backwards compatibility with wrong way to search for
+                    // multiple values (specifically, non-nested ones):
+                    if left_nl == 0 && right_nl == 1 {
+                        polars_warn!(
+                            Deprecation,
+                            "passing a list to search_sorted() for multiple values is deprecated and will be removed in Polars 2. Pass in a `Series` instead."
+                        );
+                        let other_input = expr_arena.add(AExpr::Explode {
+                            expr: input[1].node(),
+                            skip_empty: false,
```

Comment:
I think this should be `true`. It will give a `pl.Series([None])` if you are searching for nothing.

Reply:
And that implies I need more testing, I guess.